merlin/models/tf/inputs/base.py

```diff
@@ -211,68 +208,112 @@ def InputBlock(
+    categorical : Union[Tags, Schema, Layer], defaults to `Tags.CATEGORICAL`
+        A block or column-selector to use for categorical-features.
+        If a column-selector is provided (either a schema or tags), the selector
+        will be passed to `Embeddings` to infer the embedding tables from the column-selector.
+    continuous : Union[Tags, Schema, Layer], defaults to `Tags.CONTINUOUS`
+        A block to use for continuous-features.
+        If a column-selector is provided (either a schema or tags), the selector
+        will be passed to `Continuous` to infer the features from the column-selector.
     pre : Optional[BlockType], optional
         Transformation block to apply before the embeddings lookup, by default None
     post : Optional[BlockType], optional
         Transformation block to apply after the embeddings lookup, by default None
     aggregation : Optional[TabularAggregationType], optional
         Transformation block to apply for aggregating the inputs, by default "concat"
+    **branches : dict
+        Extra branches to add to the input block.
+
 
     Returns
     -------
     ParallelBlock
         Returns a ParallelBlock with a Dict with two branches:
         continuous and embeddings
     """
-    embeddings = embeddings or Embeddings(schema.select_by_tag(Tags.CATEGORICAL))
-    branches = dict(embeddings=embeddings)
+    if "categorical" not in branches:
+        cat_branch = parse_branch(categorical, Embeddings, schema)
+        if cat_branch:
+            branches["categorical"] = cat_branch
 
-    if isinstance(continuous_column_selector, Schema):
-        con_schema = continuous_column_selector
-    else:
-        con_schema = schema.select_by_tag(continuous_column_selector)
-    # TODO: Should we automatically add a Filter in TabularBlock
-    #  to filter out just the schema columns?
-    con_filter = Filter(con_schema)
-    if continuous_projection:
-        continuous = TabularBlock(schema=con_schema, aggregation="concat", pre=con_filter)
-        continuous = continuous.connect(continuous_projection)
-    else:
-        continuous = TabularBlock(schema=con_schema, pre=con_filter)
+    if "continuous" not in branches:
+        con_branch = parse_branch(continuous, Continuous, schema)
+        if con_branch:
+            branches["continuous"] = con_branch
 
-    if con_schema:
-        branches["continuous"] = continuous
+    if not branches:
+        raise ValueError("No columns selected for the input block")
 
     return ParallelBlock(
         branches,
         pre=pre,
         post=post,
         aggregation=aggregation,
     )
+
+
+def parse_branch(
+    branch: Optional[Union[Tags, Schema, Layer]],
+    default: Callable[[Schema], Layer],
+    schema: Optional[Schema] = None,
+) -> Optional[Layer]:
+    if isinstance(branch, Layer) or branch is None:
+        return branch
+
+    if isinstance(branch, Schema):
+        schema = branch
```

Comment:
Nit: Is there an edge case here where both `branch` and `schema` are provided and `schema` is overwritten? Or maybe that's intended or is okay? It looks like this function is a helper function and not intended for use by customers, so this is just a super nitty comment. If it's a helper function for internal use, maybe name it `_parse_branch`?